x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/components/add_domain/add_domain_logic.test.ts

```diff
@@ -277,32 +366,340 @@ describe('AddDomainLogic', () => {
+              await nextTick();
+
+              expect(AddDomainLogic.actions.validateDomainNetworkConnectivity).toHaveBeenCalled();
+            });
+
+            it('checks indexing restrictions after network connectivity', async () => {
+              jest.spyOn(AddDomainLogic.actions, 'validateDomainIndexingRestrictions');
+
+              AddDomainLogic.actions.performDomainValidationStep('networkConnectivity', ['url']);
+              await nextTick();
+
+              expect(AddDomainLogic.actions.validateDomainIndexingRestrictions).toHaveBeenCalled();
+            });
+
+            it('checks content after indexing restrictions', async () => {
+              jest.spyOn(AddDomainLogic.actions, 'validateDomainContentVerification');
+
+              AddDomainLogic.actions.performDomainValidationStep('indexingRestrictions', ['url']);
+              await nextTick();
+
+              expect(AddDomainLogic.actions.validateDomainContentVerification).toHaveBeenCalled();
+            });
+          });
+        });
+      });
+
+      describe('on failure', () => {
+        it('it sets all remaining validation steps as invalid', async () => {
+          http.post.mockReturnValueOnce(Promise.reject({}));
+
+          jest.spyOn(AddDomainLogic.actions, 'setDomainValidationResult');
+
+          AddDomainLogic.actions.performDomainValidationStep('initialValidation', ['url']);
+          await nextTick();
+
+          expect(AddDomainLogic.actions.setDomainValidationResult).toHaveBeenCalledWith({
+            initialValidation: {
+              state: 'invalid',
+              blockingFailure: true,
+              message: 'Unexpected error',
+            },
+            networkConnectivity: {
+              state: 'invalid',
+              message:
+                'Unable to establish a network connection because the "Initial Validation" check failed.',
+            },
+            indexingRestrictions: {
+              state: 'invalid',
+              message:
+                'Unable to determine indexing restrictions because the "Network Connectivity" check failed.',
+            },
+            contentVerification: {
+              state: 'invalid',
+              message: 'Unable to verify content because the "Network Connectivity" check failed.',
+            },
+          });
+        });
+      });
+    });
   });
 
   describe('selectors', () => {
+    describe('isValidationLoading', () => {
+      it('is false by default', () => {
+        expect(AddDomainLogic.values.isValidationLoading).toEqual(false);
+      });
+
+      it('is true when any steps are loading', () => {
+        mount({
+          domainValidationResult: {
+            steps: {
+              contentVerification: { state: 'loading' },
+              indexingRestrictions: { state: 'loading' },
+              initialValidation: { state: 'loading' },
+              networkConnectivity: { state: 'loading' },
```

Comment:
I'm updating this to add some valid states to the first few steps. As I mentioned before I'm hesitant to document states in the tests that I wouldn't expect to see in the wild. Here is an exhaustive list of possible results I expect to the logic to provide.  

```
'' '' '' ''
'loading' 'loading' 'loading' 'loading' 
'valid' 'loading' 'loading' 'loading' 
'valid' 'valid' 'loading' 'loading' 
'valid' 'valid' 'valid' 'loading'
'valid' 'valid' 'valid' 'valid' 
'invalid', 'invalid', 'invalid', 'invalid'
'valid' 'invalid' 'invalid' 'invalid'
'valid' 'valid' 'invalid' 'invalid'
'valid' 'valid' 'valid' 'invalid'
```

valid never comes ofter invalid or loading, loading never comes after invalid, we'll either have all `''` or none.

Reply:
Ahhhh sorry, I didn't mean to imply you needed to document/unit test all the various valid state combinations (although that is interesting and useful info to have!) - I was just saying it could be slightly useful to provide a 2nd example of the 4 different states in our tests, so it's not just in the type definitions of `'' | 'loading' | 'invalid' | 'valid'`. Sorry, I didn't mean it to be more annoying than that. Definitely not a blocker